content/platforms/kubernetes/db-controller.md

```diff
@@ -0,0 +1,271 @@
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: database-client
+spec:
+  containers:
+    - name: test-container
+      image: redis
+      command: [ "sleep" ]
+      args: [ "300" ]
+      env:
+      - name: REDIS_PORT
+        valueFrom:
+          secretKeyRef:
+            name: redb-mydb
+            key: port
+      - name: REDIS_HOST
+        valueFrom:
+          secretKeyRef:
+            name: redb-mydb
+            key: service_name
+      - name: REDIS_AUTH
+        valueFrom:
+          secretKeyRef:
+            name: redb-mydb
+            key: password
+
+```
+
+Then we can connect to our database via exec:
+
+```sh
+kubectl exec -it database-client -- /bin/bash -c "redis-cli -h \$REDIS_HOST -p \$REDIS_PORT -a \$REDIS_AUTH"
+```
+
+
+## Options for databases
+
+A database has some number of options that are allowed to be specified via the
+the `spec` section of the database custom resource. Some of these options are
+allowed to be changed by the user. Other options are created by the controller
+and can be consumed by using applications or developers (e.g., database
+connection information stored in a secret).
+
+### `databaseSecretName`
+
+A string containing the name of a secret that contains the desired database
+password. If this is omitted, it will default to the database name
+(metadata.name) prefixed with "redb-".
+
+This secret can be created by:
+
+```sh
+ kubectl create secret generic mydb --from-literal=password=dbpasswd
+ ```
+
+ where "`dbpasswd`" is replaced by the desired password. In the database custom
+ resource, the value is secret name (i.e., `mydb`):
+
+ ```yaml
+ databaseSecretName: mydb
+ ```
+
+ When the database is created, the secret will be updated to include the
+ port and service name for the database.
+
+### `enforceClientAuthentication`
+
+A boolean indicating whether client authentication should be enforced. The
+default is `true`.
+
+### `evictionPolicy`
+
+An eviction policy (see ["Eviction policies"](https://docs.redislabs.com/latest/rs/administering/database-operations/eviction-policy/)), defaults to `volitile-lru`.
+
+### `memorySize`
+
+The size of memory to allocate required with a minimum value of 100MB. Values
+should be specified as 1GB, 250MB, etc.
+
+### `persistence`
+
+The database persistence to use (see ["Database Persistence with Redis Enterprise Software"](https://docs.redislabs.com/latest/rs/concepts/data-access/persistence/)).
+
+The value is a keyword with the following values:
+
+| Value | Description |
+| ----- | ----------- |
+| disabled | Data is not persisted (default) |
+| AofEverySecond | Data is synced to disk every second |
+| AofAlways | Data is synced to disk with every write. |
+| SnapshotEveryHour | A snapshot of the database is created every hour |
+| SnapshotEvery6Hour | A snapshot of the database is created every 6 hours. |
+| SnapshotEvery12Hour | A snapshot of the database is created every 12 hours. |
+
+### `rackAware`
```

Comment:
Actually it defaults to the clusters setting. In a rack aware cluster it will default to `true`.

Reply:
fixed